contracts/Ink_SpokePool.sol

```diff
@@ -0,0 +1,72 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.0;
+import "@eth-optimism/contracts/libraries/constants/Lib_PredeployAddresses.sol";
+
+import "./Ovm_SpokePool.sol";
+import "./external/interfaces/CCTPInterfaces.sol";
+import { IOpUSDCBridgeAdapter } from "./external/interfaces/IOpUSDCBridgeAdapter.sol";
+
+/**
+ * @notice Ink Spoke pool.
+ * @custom:security-contact bugs@across.to
+ */
+contract Ink_SpokePool is Ovm_SpokePool {
+    using SafeERC20 for IERC20;
+
+    // Address of the custom L2 USDC bridge.
+    address private constant USDC_BRIDGE = address(0);
+
+    /// @custom:oz-upgrades-unsafe-allow constructor
+    constructor(
+        address _wrappedNativeTokenAddress,
+        uint32 _depositQuoteTimeBuffer,
+        uint32 _fillDeadlineBuffer,
+        IERC20 _l2Usdc,
+        ITokenMessenger _cctpTokenMessenger
+    )
+        Ovm_SpokePool(
+            _wrappedNativeTokenAddress,
+            _depositQuoteTimeBuffer,
+            _fillDeadlineBuffer,
+            _l2Usdc,
+            _cctpTokenMessenger
+        )
+    {} // solhint-disable-line no-empty-blocks
+
+    /**
+     * @notice Construct the OVM World Chain SpokePool.
+     * @param _initialDepositId Starting deposit ID. Set to 0 unless this is a re-deployment in order to mitigate
+     * relay hash collisions.
+     * @param _crossDomainAdmin Cross domain admin to set. Can be changed by admin.
+     * @param _withdrawalRecipient Address which receives token withdrawals. Can be changed by admin. For Spoke Pools on L2, this will
+     */
+    function initialize(
+        uint32 _initialDepositId,
+        address _crossDomainAdmin,
+        address _withdrawalRecipient
+    ) public initializer {
+        __OvmSpokePool_init(_initialDepositId, _crossDomainAdmin, _withdrawalRecipient, Lib_PredeployAddresses.OVM_ETH);
+    }
+
+    /**
+     * @notice Ink-specific logic to bridge tokens back to the hub pool contract on L1.
+     * @param amountToReturn Amount of the token to bridge back.
+     * @param l2TokenAddress Address of the l2 Token to bridge back. This token will either be bridged back to the token defined in the mapping `remoteL1Tokens`,
+     * or via the canonical mapping defined in the bridge contract retrieved from `tokenBridges`.
+     * @dev This implementation deviates slightly from `_bridgeTokensToHubPool` in the `Ovm_SpokePool` contract since World Chain has a USDC bridge which uses
+     * a custom interface. This is because the USDC token on World Chain is meant to be upgraded to a native, CCTP supported version in the future.
```

Comment:
```suggestion
     * @dev This implementation deviates slightly from `_bridgeTokensToHubPool` in the `Ovm_SpokePool` contract since
     * this chain uses Circle's bridged (upgradable to native) USDC standard, which uses a custom interface.
```